pkg/tfgen/generate.go

```diff
@@ -1009,9 +1009,11 @@ func (g *Generator) UnstableGenerateFromSchema(genSchemaResult *GenerateSchemaRe
 		}
 		files["_index.md"] = content
 	case Schema:
-		// Omit the version so that the spec is stable if the version is e.g. derived from the current Git commit hash.
-		pulumiPackageSpec.Version = ""
-
+		// For pulumi-owned (not dynamically bridged) providers, omit the version so that the spec is stable
+		// if the version is e.g. derived from the current Git commit hash.
+		if strings.Contains(g.info.Repository, "pulumi") {
+			pulumiPackageSpec.Version = ""
+		}
```

Comment:
The check doesn't do what the comment says. Specifically, it doesn't test pulumi-owned. I'm concerned that this will break external users who set `.Repository` to the wrong value (or didn't set it). Can we add restoring the version to the post-process script instead?

```patch
 dynamic/info.go | 1 +
 dynamic/main.go | 1 +
 2 files changed, 2 insertions(+)

diff --git a/dynamic/info.go b/dynamic/info.go
index d5b05fbc..76fea320 100644
--- a/dynamic/info.go
+++ b/dynamic/info.go
@@ -79,6 +79,7 @@ func providerInfo(ctx context.Context, p run.Provider, value parameterize.Value)
 				},
 				Parameter: value.Marshal(),
 			}
+			spec.Version = p.Version()
 		},
 	}
 	// Add presumed best-effort GitHub org to the provider info.
diff --git a/dynamic/main.go b/dynamic/main.go
index 0a97b970..ce8e387b 100644
--- a/dynamic/main.go
+++ b/dynamic/main.go
@@ -62,6 +62,7 @@ func initialSetup() (info.Provider, pfbridge.ProviderMetadata, func() error) {
 			spec.Attribution = ""
 			spec.Provider = schema.ResourceSpec{}
 			spec.Language = nil
+			spec.Version = version.Version()
 		},
 	}
 
```